src/main/java/org/ays/emergency_application/service/impl/EmergencyEvacuationApplicationServiceImpl.java

```diff
@@ -1,36 +1,71 @@
 package org.ays.emergency_application.service.impl;
 
 import lombok.RequiredArgsConstructor;
+import org.ays.common.model.AysPage;
+import org.ays.emergency_application.model.EmergencyEvacuationApplication;
+import org.ays.emergency_application.model.dto.request.EmergencyEvacuationApplicationListRequest;
 import org.ays.emergency_application.model.dto.request.EmergencyEvacuationApplicationRequest;
 import org.ays.emergency_application.model.entity.EmergencyEvacuationApplicationEntity;
+import org.ays.emergency_application.model.mapper.EmergencyEvacuationApplicationEntityToEmergencyEvacuationApplicationMapper;
 import org.ays.emergency_application.model.mapper.EmergencyEvacuationApplicationRequestToEntityMapper;
 import org.ays.emergency_application.repository.EmergencyEvacuationApplicationRepository;
 import org.ays.emergency_application.service.EmergencyEvacuationApplicationService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
+
 /**
  * This class implements the interface {@link EmergencyEvacuationApplicationService}
  * It is annotated with {@code @Service} to indicate that it is a service component in the application.
  * The class is also annotated with {@code @RequiredArgsConstructor} to automatically generate a constructor based on the declared final fields.
  * The {@code @Transactional} annotation ensures that all the methods in this class are executed within a transactional context.
  */
 @Service
-@Transactional
+@Transactional(readOnly = true)
```

Comment:
Bu serviste readOnly olarak verildikten sonra create işlemi de mevcut. Neden direkt olarak sınıfın üzerine yazıldı tam olarak anlayamadım.

Reply:
Spring Core kodlarında aktif kod yazan ve Hibernate üzerine çalışan Vlad Mihalcea'nın bir yazısını okumuştum. Sık bir şekilde herhangi uygulama içerisinde okuma işlemi yaptığımız için, sınıfın üzerine `@Transactional(readOnly = true)` yazmak iyi bir kullanım yöntemi_(çok özel bir durum olmadığı sürece)_ olduğundan bahsediyor. Repository'lerimizi de JPARepository'den extends ettiğimiz için `@Transactional` anotasyonunu kullanmamız Hibernate'in daha optimize çalışabiliyor. Yazının içerisindeki bir bölümü bırakıyorum, yazının totalini okursan çok daha iyi anlatılıyor.

---
> `@Transactional(readOnly = true)`, meaning that, by default, all service methods will use this setting and execute in a read-only transaction unless the method overrides the transactional settings using its own `@Transactional` definition.
For transactional services, it’s good practice to set the readOnly attribute to true at the class level and override it on a per-method basis for the service methods that need to write to the database.
https://vladmihalcea.com/spring-transactional-annotation/